src/node/stage/approved.rs

```diff
@@ -292,11 +292,17 @@ impl Approved {
     }
 
     fn progress_dkg(&mut self, core: &mut Core) {
+        let prev_has_dkg_voter = self.dkg_voter.has_live_dkg_voter();
         for (dkg_key, message) in self.dkg_voter.progress_dkg(&mut core.rng) {
             let _ = self.broadcast_dkg_message(core, &dkg_key, message);
         }
 
         self.check_dkg(core);
+
+        if prev_has_dkg_voter && !self.dkg_voter.has_live_dkg_voter() {
+            self.churn_in_progress = false;
+            self.promote_and_demote_elders(core);
+        }
```

Comment:
What if the above `check_dkg` results in a successful DKG completion? Should we still do this anyway?

Reply:
yes. but the `promote_and_demote_elders` only triggers a new DKG when there is new_elders_info (i.e. some node change happened).